pkg/distributor/distributor.go

```diff
@@ -177,6 +181,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 	flagext.DeprecatedFlag(f, "distributor.extend-writes", "Deprecated: this setting was used to try writing to an additional ingester in the presence of an ingester not in the ACTIVE state. Mimir now behaves as this setting is always disabled.", logger)
 	f.Float64Var(&cfg.InstanceLimits.MaxIngestionRate, maxIngestionRateFlag, 0, "Max ingestion rate (samples/sec) that this distributor will accept. This limit is per-distributor, not per-tenant. Additional push requests will be rejected. Current ingestion rate is computed as exponentially weighted moving average, updated every second. 0 = unlimited.")
 	f.IntVar(&cfg.InstanceLimits.MaxInflightPushRequests, maxInflightPushRequestsFlag, 2000, "Max inflight push requests that this distributor can handle. This limit is per-distributor, not per-tenant. Additional requests will be rejected. 0 = unlimited.")
+	f.Int64Var(&cfg.InstanceLimits.MaxInflightPushRequestsTotalSize, maxInflightPushRequestsTotalSizeFlag, 0, "The sum of the request sizes in bytes of inflight push requests that this distributor can handle. This limit is per-distributor, not per-tenant. Additional requests will be rejected. 0 = unlimited.")
```

Comment:
Should we make this an `int` instead of `int64` for consistency? I'm not sure it matters in reality since we only build for 64bit platforms.

Reply:
I saw we use a mixture of both for things pertaining bytes in the repo ([Int64Var](https://cs.github.com/?scopeName=All+repos&scope=&q=org%3Agrafana+repo%3Amimir+%2Fint64var%5C%28.*byte%2F) and [IntVar](https://cs.github.com/?scopeName=All+repos&scope=&q=org%3Agrafana+repo%3Amimir+%2Fintvar%5C%28.*byte%2F)) I changed to `IntVar` as this seems more common in 32c7381ac2d8cae2dcfeff304bad86f7f47999e0